File: Java/Java source-code/harmonicMean.java


public class harmonicMean {
    public static double harmonicMean_m(Double[] data) {
        double sumOfInversions = 0;
        for (int i = 0; i < data.length; i++) {
            sumOfInversions += 1 / data[i];
        }
        return data.length / sumOfInversions;
    }
}

File: Methods-C++/harmonicMean.cpp
#include <iostream>

double harmonicMean(double data[], int n) {
    double sumOfInversions = 0;
    for (int i = 0; i < n; i++) {
        sumOfInversions += 1 / data[i];
    }
    return n / sumOfInversions;
}